### ISSUE TITLE
[CouldBeSequence] Use of `asSequence` method should not be count against the `allowedOperations`

### ISSUE THREAD

**Issue by atulgpt** [detekt/detekt#8776]
## Expected Behavior
```kt
// allowed function call config is set to 2
private val subject = CouldBeSequence(TestConfig("allowedOperations" to 2))
@Test
fun `using asSequence after allowed number of operations should not trigger rule`() {
    val code = """
        fun foo() = listOf<String?>("bar")
            .mapIndexed { index, it ->  it?.let { it + "_" + index } }
            .filterNotNull()
            .asSequence()
    """.trimIndent()
    assertThat(subject.lintWithContext(env, code)).isEmpty()
}
```
above TC should pass

## Observed Behavior
Above mentioned TC is failing

## Steps to Reproduce
Run against the latest main branch

## Your Environment

* Version of detekt used: Latest main branch(00ccb8c3691089c40529a2ac510c1ce3ffbc23f8) of the detekt
